packages/streamline/src/components/inputs/input-text/components/input-text-primitive.tsx

```diff
@@ -0,0 +1,188 @@
+import { Box } from '../../../../primitives/box/box';
+import { Icon } from '../../../../primitives/icon/icon';
+import { Text } from '../../../../primitives/text/text';
+import { ColorTheme, useStreamlineTheme } from '../../../../theme';
+import React from 'react';
+import {
+  TextInput as NativeTextInput,
+  TextInputProps,
+  View,
+} from 'react-native';
+import { IconAdornment } from './input-text-icon';
+import { InputTextLabel } from './input-text-label';
+import { InputTextOutline, OUTLINE_WIDTH } from './input-textoutline';
+
+import { TextInputPrimitiveProps } from './types';
+
+const TEXT_INPUT_HEIGHT_WITHOUT_LABEL = 48;
+const DEFAULTLINEHEIGHT = 18;
+
+export const InputTextPrimitive = ({
+  isSearchInput,
+  isDisabled,
+  isEditable,
+  isError,
+  description,
+  multiline,
+  numberOfLines = 1,
+  isFocused,
+  label,
+  placeholder,
+  secureTextEntry,
+  value,
+  onFocus,
+  onBlur,
+  onChangeText,
+  right,
+  left,
+  forceFocus,
+  parentState,
+  innerRef,
+  pointerEvents,
+  render = (props: TextInputProps) => <NativeTextInput {...props} />,
+  maxWidth,
+  width,
+  height,
+  noPadding,
+  ...rest
+}: TextInputPrimitiveProps) => {
+  const theme = useStreamlineTheme();
+  const gutterStyle = { marginLeft: left ? theme.spacing.xs : 0 };
+  const containerStyle = !label &&
+    !isSearchInput && { paddingTop: theme.spacing.xxs };
+  const inputPadding = { paddingVertical: theme.spacing.xs + OUTLINE_WIDTH };
+  const inputContainer = (
```

Comment:
Open question, should it be its own component to make it more easily readable?

Reply:
I understand your concern but I don't think it will make it more readable. Better to have all in the same render